tuist/tuist
# Support for custom Project.Options for swift packages in Dependencies.swift

Resolves https://github.com/tuist/tuist/issues/4448

### Short description 📝

Fix a bug which was stopping tuist from creating automatic resource accessors. This PR also allows users to control resource accessors generation by defining `Project.Options` in SwiftPackageManagerDependencies object in `Dependencies.swift`

### How to test the changes locally 🧐

1. Point `tuist` to a fixture that have resources in a swift package. Swift package should use default SPM resources i.e 

https://developer.apple.com/documentation/swift_packages/bundling_resources_with_a_swift_package
https://developer.apple.com/documentation/swift_packages/localizing_package_resources

2. Run `tuist` using Fourier. 
3. Generated project should have resource accessors generated for SPM modules4. 

### Checklist ✅
